tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/config/ITmfConfigurationSourceType.java

```diff
@@ -49,4 +52,16 @@ public interface ITmfConfigurationSourceType {
      * @return A list of query parameter descriptors to be passed
      */
     List<ITmfConfigParamDescriptor> getConfigParamDescriptors();
+    /**
+     * A string containing a json-schema describing the query parameters
+     * to be passed when creating a configuration instance of this type.
+     *
+     * Note: Use either {@link #getConfigParamDescriptors()} or {@link #getSchemaFile()}
```

Comment:
For the user, they don't know which is to be used. I would suggest:
Note: If the schema file is null, use {@link #getConfigParamDescriptors()} instead.

Should we also add a note in that other method?
Note: Use this method if {@link #getSchemaFile()} returns null.